_ap/18/index.adoc

```diff
@@ -3,76 +3,194 @@ num: 18
+
+== Proposed Architecture
+
+We propose to switch to alternative OSS tooling provided by Microsoft called https://github.com/microsoft/kiota[Kiota].
+
+=== Advantages:
+
+* Community
+* Better design/architecture
+* Better developer experience
+* Maintenance
+* Swappable implementation libraries
+* Compatibility matrix
+
+=== Disadvantages:
+
+* Not GA (yet)
+* External Owners (Microsoft)
+* Core Generator codebase is .NET
+
+==== Community
+
+The responses in the community are quick and provide context https://github.com/microsoft/kiota-java/issues/121#issuecomment-1372248993[above expectations].
+
+Collaboration is actively fostered and contributing to both the core generator and the auxiliary libraries is facilitated and encouraged with active and timely feedback.
+
+Cross-project integrations are facilitated and https://github.com/microsoft/kiota/discussions/2154[healthy conversations] are initiated on both (RH and Microsoft) sides.
+
+==== Better design/architecture
+
+Kiota design is extremely principled and follows https://microsoft.github.io/kiota/extending/[good and detailed guidelines] throughout the project.
+The project leadership shows broad and deep experience in building successful code generators, which are pragmatic and extensible.
+
+==== Better developer experience
+
+The project's architectural design decisions are supported and embraced throughout all of the different implementations so that the user is going to have a consistent and uniform experience while using the generated code across different languages.
+Additionally, the fluent code style of the generated clients provides a better experience for consumers, resulting in higher levels of developer joy.
+
+==== Maintenance
+
+Experience shows that issues are answered within hours, PRs are continuously accepted and merged, and guidance is provided when needed.
+
+Transitive dependencies are kept up to date in a timely manner.
+
+Improvements and bugs are continuously integrated into both the generator and the auxiliary libraries' repositories.
+
+==== Swappable implementation libraries
+
+The design of Kiota allows consumers of the code generator to provide their own https://microsoft.github.io/kiota/extending/corelibrary.html[“opinionated” implementations] of core generated client functionality.
+This is done via a set of interfaces to core functionality that all generated clients leverage (e.g. a Java HTTP client interface).  Default implementations of these interfaces are provided, but alternatives can easily be created and used.
+
+Specifically for the Red Hat use case, we are free to roll out our own implementations of the core libraries to better support Red Hat products and strategy (e.g. Quarkus specific implementations of certain functionality).
```

Comment:
It would be good to spell out the implementations we're expecting to write. HTTP client is one, but what about authentication, for example?

Reply:
Good question, regarding implementations, for Java I'd expect:
- a Vert-X HTTP client implementation (as opposed to the default OkHttp)
- a JSON serialization based on Jackson (as opposed to the default Gson)

Do we have strong opinions on the other languages @tombentley ?

@riccardo-forina can you check the TS libraries? https://github.com/microsoft/kiota-typescript
@dimakis can you have a look over Python? https://github.com/microsoft?q=kiota+python&type=all&language=&sort=
@lburgazzoli can I ask for your support in Go? https://github.com/microsoft?q=kiota+go&type=all&language=&sort=

🙏  thanks in advance!

Regarding Authentication I think that we should support only OFFLINE_TOKEN, this means rolling out a custom `TokenProvider` for each language (it's not mandatory but highly conveninet). this is in Java for example:
https://github.com/andreaTP/kiota-utils#redhat-sso-auth

Reply:
@tombentley addressed here: https://github.com/bf2fc6cc711aee1a0c2a/architecture/pull/90/commits/c9f2cee1fc624782620d4ca59498b5bf64c50ca1 let me know if you prefer additional detail.